lib/mlton/basic/hash.sig

```diff
@@ -0,0 +1,21 @@
+(* Copyright (C) 2017 Jason Carr
+ *
+ * MLton is released under a HPND-style license.
+ * See the file MLton-LICENSE for details.
+ *)
+
+(* HASH defines a set of transformations on hash values *)
+signature HASH =
+   sig
+      (* combine a list of hash values so that each is independent from the last.
+       * No guarantee is made as to whether or not combine [w] = w *)
+      val combine: word List.t -> word
```

Comment:
I'm hesitant to do this given the current usage. Every usage I have for Hash.combine is with a constant-sized list of a few hashes, making use of list literals. So although it's reasonable to do so, the semantics I have for it here treat it more as a variadic function instead of a function on lists.

Reply:
Then maybe we want
```
  val combine2 : word * word -> word
  val combine3 : word * word * word -> word
  val list : word List.t -> word
```
The `combine<N>` functions would generally be more efficient, because MLton won't always unroll iterations on list literals.

Reply:
Oh, I had expected it to. I'll make that change then.